paymentsheet/src/main/java/com/stripe/android/paymentsheet/navigation/PaymentSheetScreen.kt

```diff
@@ -77,9 +82,49 @@ internal sealed interface PaymentSheetScreen {
 
     fun topBarState(): StateFlow<PaymentSheetTopBarState?>
 
-    fun title(isCompleteFlow: Boolean, isWalletEnabled: Boolean): StateFlow<ResolvableString?>
-
-    fun showsWalletsHeader(isCompleteFlow: Boolean): StateFlow<Boolean>
+    fun title(isCompleteFlow: Boolean, isWalletEnabled: Boolean): ResolvableString?
+
+    fun showsWalletsHeader(isCompleteFlow: Boolean, walletsState: WalletsState?): Boolean
+
+    /**
+     * State Dependency Graph:
+     *
+     *     walletState      currentScreen     primaryButtonUiState
+     *       \              |         |              |
+     *        \             v         |              |
+     *         \   showsWalletsHeader |              |
+     *          \          |          |              |
+     *           v         v          v              |
+     *          actualWalletsState -> headerText     |
+     *                    \          /               |
+     *                     v        v                v
+     *                  PaymentSheetScreenContentState
+     *
+     */
+    fun paymentSheetScreenContentState(
```

Comment:
I think there was a misunderstanding here. What I meant was for this to be abstract, and for each of the screens to implement it. I also meant we should remove the other StateFlows and use only this one.

Reply:
As is, this implementation has the same issues mentioned [here](https://github.com/stripe/stripe-android/pull/10524#discussion_r2024808838).